keyboards/mincedshon/ecila/config.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 mincedshon (@mincedshon)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/*Bootmagic boot button, set to topmost, leftmost key */
+#define BOOTMAGIC_LITE_ROW	0
+#define BOOTMAGIC_LITE_COLUMN	0
+
+/* encoder */
+/* #define NUMBER_OF_ENCODERS 1 */
+//#define ENCODERS_PAD_A { D3 }
+//#define ENCODERS_PAD_B { D2 }
+//#define ENCODER_RESOLUTION 1
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 15
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { E6, B3, B5, B4, D7 }
+#define MATRIX_COL_PINS { B6, F7, F6, F5, F4, F1, F0, D6, B7, D0, D1, D2, D3, D5, D4 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
```

Comment:
IF you want, you can specify these in info.json as well.

Reply:
may i know the reason behind this?? am a little lost on why the IDs need to be moved too

Reply:
because we're moving to a data driven model, where most of the info is in the info.json.

Reply:
ah okay, will do so!

Reply:
do lines 18-31 translate to this in info.json?

    "MATRIX_ROWS": 5,
    "MATRIX_COLS": 15,
    "MATRIX_ROW_PINS": { E6, B3, B5, B4, D7 },
    "MATRIX_COL_PINS": { B6, F7, F6, F5, F4, F1, F0, D6, B7, D0, D1, D2, D3, D5, D4 },
    "UNUSED_PINS": ,
    "DIODE_DIRECTION": COL2ROW,
    "DEBOUNCE": 5,

Reply:
https://docs.qmk.fm/#/reference_info_json

Reply:
thank you!